homeassistant/components/nad/media_player.py

```diff
@@ -185,23 +185,23 @@ def source_list(self):
         """List of available input sources."""
         return sorted(list(self._reverse_mapping.keys()))
 
-    def update(self):
+    def update(self) -> None:
         """Retrieve latest state."""
-        if self._nad_receiver.main_power("?") == "Off":
-            self._state = STATE_OFF
-        else:
-            self._state = STATE_ON
-
-        if self._nad_receiver.main_mute("?") == "Off":
-            self._mute = False
-        else:
-            self._mute = True
+        power_state = self._nad_receiver.main_power("?")
+        if not power_state:
+            self._state = None
```

Comment:
When `self._state` is `None`, shouldn't we also return `False` from the `unavailable property? We don't know if it's on/off, the device is unavailable.